core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -155,39 +167,87 @@ protected List<String> scanColumns() {
+     *       contribute to the residuals of tasks returned by this scan. (See {@link
+     *       PositionDeletesScanTask#residual()})
+     * </ul>
+     *
+     * @param expr expression filter that applies on the base table of this posiiton deletes table
+     * @return this for method chaining
+     */
+    public BatchScan baseTableFilter(Expression expr) {
+      return new PositionDeletesBatchScan(
+          table(), schema(), context(), Expressions.and(baseTableFilter, expr));
+    }
+
+    /**
+     * Sets a filter to use for this scan.
+     *
+     * <p>This must be set exclusively from {@link #baseTableFilter(Expression)}
+     *
+     * @param expr expression filter
+     * @return this for method chaining
+     */
+    @Override
+    public BatchScan filter(Expression expr) {
+      return new PositionDeletesBatchScan(
+          table(),
+          schema(),
+          context().filterRows(Expressions.and(context().rowFilter(), expr)),
+          baseTableFilter);
+    }
+
     @Override
     protected CloseableIterable<ScanTask> doPlanFiles() {
       String schemaString = SchemaParser.toJson(tableSchema());
 
       // prepare transformed partition specs and caches
       Map<Integer, PartitionSpec> transformedSpecs = transformSpecs(tableSchema(), table().specs());
 
+      LoadingCache<Integer, String> specStringCache =
+          partitionCacheOf(transformedSpecs, PartitionSpecParser::toJson);
+      LoadingCache<Integer, ManifestEvaluator> deletesTableEvalCache =
+          partitionCacheOf(
+              transformedSpecs,
+              spec -> ManifestEvaluator.forRowFilter(filter(), spec, isCaseSensitive()));
+      LoadingCache<Integer, ManifestEvaluator> baseTableEvalCache =
+          partitionCacheOf(
+              table().specs(), // evaluate base table filters on base table specs
+              spec -> ManifestEvaluator.forRowFilter(baseTableFilter, spec, isCaseSensitive()));
       LoadingCache<Integer, ResidualEvaluator> residualCache =
           partitionCacheOf(
               transformedSpecs,
               spec ->
                   ResidualEvaluator.of(
                       spec,
+                      // there are no applicable filters in the base table's filter
+                      // that we can use to evaluate on the position deletes table
                       shouldIgnoreResiduals() ? Expressions.alwaysTrue() : filter(),
                       isCaseSensitive()));
 
-      LoadingCache<Integer, String> specStringCache =
-          partitionCacheOf(transformedSpecs, PartitionSpecParser::toJson);
-
-      LoadingCache<Integer, ManifestEvaluator> evalCache =
-          partitionCacheOf(
-              transformedSpecs,
-              spec -> ManifestEvaluator.forRowFilter(filter(), spec, isCaseSensitive()));
-
       // iterate through delete manifests
       List<ManifestFile> manifests = snapshot().deleteManifests(table().io());
 
       CloseableIterable<ManifestFile> matchingManifests =
```

Comment:
Shall we do this filter only if either of the filters is non-trivial? Otherwise, what's the point of doing this work?

Reply:
Well, this is for manifests, so shouldn't matter that much. Never mind.